sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -565,6 +565,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
       filters: Array[Filter],
       inputPaths: Array[String],
       broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
@liancheng seems refresh the file status is unavoidable. let's do that right before getting the input files.